src/microservices/Microservices.CohortExtractor/Execution/RequestFulfillers/FromCataloguesExtractionRequestFulfiller.cs

```diff
@@ -79,6 +81,41 @@ public IEnumerable<ExtractImageCollection> GetAllMatchingFiles(ExtractionRequest
             }
         }
 
+        public IEnumerable<IRejector> GetRejectorsFor(ExtractionRequestMessage message, QueryToExecute query)
+        {
+            if (message.IsNoFilterExtraction)
+            {
+                return Enumerable.Empty<IRejector>();
+            }
+
+            if(ModalitySpecificRejectors.Any()  && string.IsNullOrWhiteSpace(query.Modality))
+                throw new Exception("Could not evaluate ModalitySpecificRejectors because query Modality was null");
```

Comment:
If I'm reading this correctly - if any modality rejectors are defined in the config, then we require that the modality is always specified?

Reply:
`query` is an SQL query that is to be run on a table.  This is always calculated from the table name e.g. CT_ImageTable = CT etc.  The pattern used to pull out the modality is determined by `ModalityMatchingRegex` regardless of what the RabbitMQ message was for.  `ModalityMatchingRegex` is optional, you can just stick all images in one table after all.

This line says 'if you haven't told me how to extract Modality from Catalogue names then I can't apply custom modality specific rejection rules'

We use the name to determine the modality in order to avoid querying the data itself which might not even have a Modality field in it (e.g. the ImageTables typically don't)